### ISSUE TITLE
Provide cmake-package

### ISSUE THREAD

**Issue by cNoNim** [Dav1dde/glad#452]
Hello,

Maintaining [conda-forge/glad-feedstock](https://github.com/conda-forge/glad-feedstock) is my goal (https://github.com/conda-forge/glad-feedstock/issues/7).

But I have an idea how to improve CMake support in upstream.
What do you think about translating [cmake/CMakeLists.txt](https://github.com/Dav1dde/glad/blob/glad2/cmake/CMakeLists.txt) into GladConfig.cmake?

To ensure backward compatibility, we can set CMAKE_PREFIX_PATH to a new location of GladConfig.cmake and enable the use of python module from sources inside cmake/CMakeLists.txt.

If it's okay with you, I can provide PR.

**Comment by Dav1dde** [comment]
Thanks for the all the work you and all the other people do for packaging glad.

Unfortunately I have pretty much no idea about cmake and it has always been a pain for me to keep it up to date and usable, I always relied on community help for that.

Is this change something you need in the repo to make the packaging possible? I would prefer if the packaging was as much as possible independent from the glad repo.

That being said, if you can keep the changes backwards compatible and it improves the overall experience I am thankful for any contribution.

**Comment by cNoNim** [comment]
@Dav1dde you can review PR https://github.com/Dav1dde/glad/pull/453 is it ready
It's actually not required.
I already prepared PR for conda-forge https://github.com/conda-forge/glad-feedstock/pull/8

But it seems to me that such a solution will simplify the overall experience.

After install python module and copy GladConfig.cmake
u just need
```cmake
find_package(Glad REQUIRED)
glad_add_library(glad_gl_core_33 REPRODUCIBLE API gl:core=3.3)
```

**Comment by cNoNim** [comment]
Actually, for packaging u just need

patch CMakeLists.txt -> [cmake.patch](https://github.com/conda-forge/glad-feedstock/blob/c3d6fe3e0aff3680707332540b0fcbdfaf3cb63f/recipe/0001-cmake.patch)

```shell
mkdir -p $PREFIX/lib/cmake/Glad && cp $SRC_DIR/cmake/CMakeLists.txt $PREFIX/lib/cmake/Glad/GladConfig.cmake
```

**Comment by cNoNim** [comment]
@Dav1dde 

Sorry I don't want to create a separate issue for a small question.
It looks like there are some interesting CMake level improvements from the last release.
When will 2.0.5 be released?

**Comment by Dav1dde** [comment]
Thanks for your PR, I'll take a look on the weekend and make a new release for you.

**Comment by cNoNim** [comment]
It's not urgent. And thanks for your work.
Maybe we need to look deeper into the cmake packages.
For example, if you look at https://github.com/Tradias/asio-grpc/blob/master/cmake/asio-grpcConfig.cmake.in it looks more complicated, but I don't really understand what it's so complicated for.

**Comment by Dav1dde** [comment]
Thanks for your work, merged and released 2.0.5!

**Comment by cNoNim** [comment]
Thanks!
